django_pg_simple_hll/migrations/0002_custom_hashing.sql

```diff
@@ -0,0 +1,106 @@
+-- drop some methods that we will overwrite
+DROP AGGREGATE IF EXISTS hll_cardinality(anyelement, int);
+DROP AGGREGATE IF EXISTS hll_cardinality(anyelement);
+DROP FUNCTION IF EXISTS hll_bucket(int [], anyelement);
+DROP FUNCTION IF EXISTS hll_bucket(int [], anyelement, int);
+
+
+-- Default hashing function, it turns any input into an unsigned 32 bit integer
+CREATE OR REPLACE FUNCTION hll_hash(input anyelement) RETURNS int
+LANGUAGE sql IMMUTABLE STRICT PARALLEL SAFE AS $$
+SELECT HASHTEXT(input::text) & ~(1 << 31)
+$$;
+
+-- The state transition function
+-- it takes the current state, and the hashed input values
+-- hll_agg_state is the current running state, made up of bucket keys and hashes
+--  it is an array of n_buckets
+-- hashed_input is the int32 hash of any element we are considering
+-- hll_precision is the precision we are using for the approximation
+--  it use used to calculate the number of buckets
+CREATE OR REPLACE FUNCTION hll_bucket(
+    hll_agg_state int [],
+    hashed_input int,
+    hll_precision int
+) RETURNS int []
+LANGUAGE plpgsql IMMUTABLE STRICT PARALLEL SAFE AS $$
+DECLARE
+    -- eg. for 512 buckets, hll_precision is 9 because 2**9 -> 512
```

Comment:
this reads a bit strangely.  Since `hll_precision` is the agrument that varies, should this be "for hll_precision 9, n_buckets is 512 because 2**9 -> 512"